consensus/propagation/have_wants.go

```diff
@@ -317,9 +263,28 @@ func (blockProp *Reactor) handleRecoveryPart(peer p2p.ID, part *proptypes.Recove
 		return
 	}
 
+	// todo: add these defensive checks in a better way
+	if cb == nil {
+		return
+	}
+	if parts == nil {
+		return
+	}
+
+	// todo: we need to figure out a way to get the proof for a part that was
+	// sent during catchup.
+	proof := cb.GetProof(part.Index)
+	if proof == nil {
+		if part.Proof == nil {
+			blockProp.Logger.Error("proof not found", "peer", peer, "height", part.Height, "round", part.Round, "part", part.Index)
+			return
+		}
+		proof = part.Proof
+	}
```

Comment:
now we get proofs that are stored in the compact block instead of including them in messages. now proofs are only included during catchup